### PR TITLE
HistoryIntervalStopFoundEvent passes the message CheckoutHistoryActivity to the message bus

### PR DESCRIPTION
**Pull request by donabp**

### CODE REVIEW

File: Corgibytes.Freshli.Cli/Functionality/History/HistoryIntervalStopFoundEvent.cs [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+    public string? GitExecutablePath { get; init; }
 
     public void Handle(IApplicationActivityEngine eventClient)
     {
+        if (AnalysisLocation != null && GitExecutablePath != null
+                                     && AnalysisLocation.CommitId != null)
+        {
+            eventClient.Dispatch(new CheckoutHistoryActivity(
+                eventClient.ServiceProvider.GetRequiredService<IGitManager>(), GitExecutablePath,
+                AnalysisLocation.CacheDirectory, AnalysisLocation.RepositoryId, AnalysisLocation.CommitId));
```
Comment (klhoot) [inline Corgibytes.Freshli.Cli/Functionality/History/HistoryIntervalStopFoundEvent.cs:22]: Could we pass AnalysisLocation to CheckoutHistoryActivity as a single parameter?

Comment (donabp) [reply Corgibytes.Freshli.Cli/Functionality/History/HistoryIntervalStopFoundEvent.cs:22]: I can do that and I just tried it, I just need to adjust the unit test file as well. on it :)

**Comment by mscottford** [comment]
Which issue does this pull request address? Please put a "Fixes __" line in the description so it's correctly linked with that issue in the project board.

**Comment by donabp** [comment]
Fixes #208 

This spawns from https://github.com/corgibytes/freshli-cli/pull/207
Blocked by this PR: https://github.com/corgibytes/freshli-cli/pull/204/files#diff-8b3691b75312db3758ae2ac096f46f86676c6534002d0cb8832fd462933b4de0

 HistoryIntervalStopFoundEvent.Handle
 Passes the message CheckoutHistoryActivity to the message bus

**Pull request merged**